internal/contracts/operations.go

```diff
@@ -73,6 +73,21 @@ func (cm *contractManager) RunOperation(ctx context.Context, op *fftypes.Prepare
 }
 
 func (cm *contractManager) OnOperationUpdate(ctx context.Context, op *fftypes.Operation, update *operations.OperationUpdate) error {
+	// Special handling for OpTypeBlockchainInvoke, which writes an event when it succeeds or fails
+	if op.Type == fftypes.OpTypeBlockchainInvoke {
+		if update.Status == fftypes.OpStatusSucceeded {
+			event := fftypes.NewEvent(fftypes.EventTypeBlockchainInvokeOpSucceeded, op.Namespace, op.ID, op.Transaction, "")
```

Comment:
Note that I am not setting a `topic` or `correlator` on this event. Open to thoughts if there's anything sensible to be used for either. All we have is the operation (and therefore the inputs to the "invoke" request).

Reply:
I don't have and great solutions.
For blockchain events, we use the specified topic, then fall back to the listener ID. Special case for the built-in listener.

https://github.com/hyperledger/firefly/blob/0cdaa9747040fb6ee07683464b1c21463dbf29c4/internal/events/blockchain_event.go#L72-L87

I don't see any obvious parallels.
My best suggestion in terms of usefulness on the `topic` is the first 64 characters of the query-stringified `location`.
Whether or not that's good enough to justify the cost of extracting it from the input, I don't know.

Reply:
... and I realize it would need sanitization to fit the string requirements of the topic I think

Reply:
Yea - extracting, truncating, and sanitizing that location seems like a lot to me. Going to leave it for now and we can revisit if needed.